node/tests/testcase.go

```diff
@@ -262,60 +263,71 @@ func testMarshal(t *testing.T, n datamodel.Node, data string) {
 	if err != nil {
 		t.Errorf("marshal failed: %s", err)
 	}
-	Wish(t, buf.String(), ShouldEqual, reformat(data, prettyprint))
+	qt.Check(t, buf.String(), qt.Equals, reformat(data, prettyprint))
 }
 
 func wishPoint(t *testing.T, n datamodel.Node, point testcasePoint) {
 	t.Helper()
 	reached, err := traversal.Get(n, datamodel.ParsePath(point.path))
 	switch point.expect.(type) {
 	case error:
-		Wish(t, err, ShouldBeSameTypeAs, point.expect)
-		Wish(t, err, ShouldEqual, point.expect)
+		qt.Check(t, err, qt.ErrorAs, point.expect)
+		qt.Check(t, err, qt.Equals, point.expect)
 	default:
-		Wish(t, err, ShouldEqual, nil)
+		qt.Check(t, err, qt.Equals, nil)
 		if reached == nil {
 			return
 		}
-		Wish(t, reached, closeEnough, point.expect)
+		qt.Check(t, reached, closeEnough, point.expect)
 	}
 }
 
-// closeEnough conforms to wish.Checker (so we can use it in Wish invocations),
+// closeEnough conforms to quicktest.Checker (so we can use it in quicktest invocations),
 // and lets Nodes be compared to primitives in convenient ways.
 //
 // If the expected value is a primitive string, it'll AsStrong on the Node; etc.
 //
 // Using a datamodel.Kind value is also possible, which will just check the kind and not the value contents.
 //
-// If a datamodel.Node is the expected value, a full deep ShouldEqual is used as normal.
-func closeEnough(actual, expected interface{}) (string, bool) {
+// If a datamodel.Node is the expected value, a full deep qt.Equals is used as normal.
+var closeEnough = &closeEnoughChecker{}
+
+var _ qt.Checker = (*closeEnoughChecker)(nil)
```

Comment:
clever :) I like how this turned out, even if implementing a custom checker with quicktest is slightly more verbose.

you could almost use qt.CmpEquals with https://pkg.go.dev/github.com/google/go-cmp/cmp#Comparer here, except that Comparer expects `func(T, T) bool`, so it only kicks in when the two types being compared are exactly equal. The logic here is different, as it is more flexible.